src/rust/engine/workunit_store/src/lib.rs

```diff
@@ -128,6 +129,7 @@ pub struct WorkunitMetadata {
   pub stdout: Option<hashing::Digest>,
   pub stderr: Option<hashing::Digest>,
   pub artifacts: Vec<(String, hashing::Digest)>,
+  pub user_metadata: Vec<(String, Arc<PyObject>)>,
```

Comment:
why call it "user_metadata" instead of "metadata" ?
do we want to allow for other metadata files ? (e.g. engine_metadata) ?

Reply:
We already have a  struct called `WorkunitMetadata`. I wanted to make sure the user-supplied metadata was called something different in Rust code. It might be worth it to rename that existing Rust struct, but that's out of scope for this PR I think.